stock_request/models/stock_request.py

```diff
@@ -235,14 +235,23 @@ def action_draft(self):
 
     def action_cancel(self):
         self.sudo().mapped("move_ids")._action_cancel()
-        self.write({"state": "cancel"})
+        self._action_cancel()
         return True
 
+    def _action_cancel(self):
+        self.write({"state": "cancel"})
```

Comment:
Why an extra method?

Reply:
It is not really necessary, but by using the same technique as the `action_done()` method.